examples/idle.rs

```diff
@@ -0,0 +1,77 @@
+use imap::extensions::idle;
+use native_tls::TlsConnector;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "idle")]
+struct Opt {
+    // The server name to connect to
+    #[structopt(short, long)]
+    server: String,
+
+    // The port to use
+    #[structopt(short, long, default_value = "993")]
+    port: u16,
+
+    // The account username
+    #[structopt(short, long)]
+    username: String,
+
+    // The account password. In a production system passwords
+    // would normally be in a config or fetched at runtime from
+    // a password manager or user prompt and not passed on the
+    // command line.
+    #[structopt(short = "w", long)]
+    password: String,
+
+    // The mailbox to IDLE on
+    #[structopt(short, long, default_value = "INBOX")]
+    mailbox: String,
+
+    #[structopt(
+        short = "x",
+        long,
+        help = "The number of responses to receive before exiting",
+        default_value = "5"
+    )]
+    max_responses: usize,
```

Comment:
Hmm.. I get what you're saying about complexity, but let me try to convince you that it is fine as is? If I am unsuccessful, then I don't really object to making these const.

- These options all already have sensible default values, so you don't need to specify them.
- It is more consistent to have everything passed into the imap client config come from the same place.
- Having these as options is consistent with what someone who is cribbing from the example is going to do anyway.
- It is not really any more code to have these as options, since having them as options is just a variable declaration and a structopt macro, instead of a const decl and maybe a comment.

Reasonable? :)

Reply:
I'm sold :)